types/hapi/definitions/request/request-events.d.ts

```diff
@@ -0,0 +1,45 @@
+import * as Podium from "podium";
+
+/**
+ * 'peek' - emitted for each chunk of payload data read from the client connection. The event method signature is function(chunk, encoding).
+ * 'finish' - emitted when the request payload finished reading. The event method signature is function ().
+ * 'disconnect' - emitted when a request errors or aborts unexpectedly.
+ * For context [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-requestevents)
+ */
+export type RequestEventType = "peek" | "finish" | "disconnect";
+
+/**
+ * Access: read only and the public podium interface.
+ * The request.events supports the following events:
+ * * 'peek' - emitted for each chunk of payload data read from the client connection. The event method signature is function(chunk, encoding).
+ * * 'finish' - emitted when the request payload finished reading. The event method signature is function ().
+ * * 'disconnect' - emitted when a request errors or aborts unexpectedly.
+ * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-requestevents)
+ */
+export interface RequestEvents extends Podium {
+
+    /**
+     * Access: read only and the public podium interface.
+     * The request.events supports the following events:
+     * * 'peek' - emitted for each chunk of payload data read from the client connection. The event method signature is function(chunk, encoding).
+     * * 'finish' - emitted when the request payload finished reading. The event method signature is function ().
+     * * 'disconnect' - emitted when a request errors or aborts unexpectedly.
+     * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-requestevents)
+     */
+    on(criteria: RequestEventType, listener: Function): void;
```

Comment:
You can use a mapped type to declare all `on/once` methods on this interface.

Reply:
@SimonSchick can you give me more info please? I read about Mapped Type ([link-1](https://www.typescriptlang.org/docs/handbook/advanced-types.html), [link-2](https://blog.mariusschulz.com/2017/01/20/typescript-2-1-mapped-types)) but I can't see how to use in this case. We are overriding methods (on/once) from Podium interface but with some differents parameters. How can I use Mapped Type in this case please? Thanks for the feedback!

Reply:
https://github.com/SimonSchick/EliteDangerousUtils/blob/master/src/EDLog/EDLog.ts#L307